### PR TITLE
Feature/weee reused site list

### PR DESCRIPTION
**Pull request by JB12298**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/Mappings/AatfSiteMap.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿namespace EA.Weee.RequestHandlers.Mappings
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Domain.AatfReturn;
+    using Aatf = Core.AatfReturn.AatfData;
+
+    public class AatfSiteMap : IMap<AatfAddressObligatedAmount, AddressTonnageSummary>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Mappings/AatfSiteMap.cs]: Can we add some tests for this :)

File: src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReusedOffSiteSummaryListController.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        [HttpGet]
+        public virtual async Task<ActionResult> Index(Guid organisationId, Guid returnId, Guid aatfId)
+        {
+            var viewModel = new ReusedOffSiteSummaryListViewModel();
+            var returnViewModel = new ReturnViewModel();
+            using (var client = apiClient())
+            {
+                var sites = await client.SendAsync(User.GetAccessToken(), new GetAatfSite(aatfId, returnId));
+                viewModel = mapper.Map<ReusedOffSiteSummaryListViewModel>(sites);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReusedOffSiteSummaryListController.cs]: Remove this line

File: src/EA.Weee.Web.Tests.Unit/Areas/AatfReturn/Controller/ReusedOffSiteSummaryListControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        [Fact]
+        public async void IndexGet_GivenValidViewModel_BreadcrumbShouldBeSet()
+        {
+            var organisationId = Guid.NewGuid();
+
+            await controller.Index(organisationId, A.Dummy<Guid>(), A.Dummy<Guid>());
+
+            Assert.Equal(breadcrumb.ExternalActivity, BreadCrumbConstant.AatfReturn);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/AatfReturn/Controller/ReusedOffSiteSummaryListControllerTests.cs]: Add asserts that the ExternalOrganisation and SchemePublicInfo properties are set on the breadcrumb

File: src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/AddressUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            address = addressData.Address2 == null ? address : StringConcatenate(address, addressData.Address2);
+
+            address = StringConcatenate(address, addressData.TownOrCity);
+
+            address = addressData.CountyOrRegion == null ? address : StringConcatenate(address, addressData.CountyOrRegion);
+
+            address = addressData.Postcode == null ? address : StringConcatenate(address, addressData.Postcode);
+
+            address = StringConcatenate(address, addressData.CountryName);
+
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/AddressUtilities.cs]: Add unit tests around this, i.e. create a AddressUtilitiesTests class

File: src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/TonnageUtilities.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
+﻿namespace EA.Weee.Web.Areas.AatfReturn.Mappings.ToViewModel
+{
+    using System.Collections.Generic;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Core.Helpers;
+
+    public class TonnageUtilities
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/TonnageUtilities.cs]: Create an interface for this so we can inject it into any classes that need it

File: src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToReturnViewModelMap.cs [hunk 6]
```diff
@@ -14,6 +14,7 @@ public class ReturnToReturnViewModelMap : IMap<ReturnData, ReturnViewModel>
         public decimal? NonObligatedTonnageTotalDcf = null;
 
         public List<AatfObligatedData> AatfObligatedData = new List<AatfObligatedData>();
+        private TonnageUtilities tonnageUtilities = new TonnageUtilities();
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToReturnViewModelMap.cs]: This should be injected in the constructor, public ReturnToReturnViewModelMap(ITonnageUtilities tonnageUtilities)

File: src/EA.Weee.RequestHandlers/Mappings/AatfSiteMap.cs (hunk 1 above)
Comment (JB12298) [reply src/EA.Weee.RequestHandlers/Mappings/AatfSiteMap.cs]: Done

File: src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReusedOffSiteSummaryListController.cs (hunk 2 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReusedOffSiteSummaryListController.cs]: Moved

File: src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToReturnViewModelMap.cs (hunk 6 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToReturnViewModelMap.cs]: Done

File: src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/TonnageUtilities.cs (hunk 5 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/TonnageUtilities.cs]: Done

File: src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/AddressUtilities.cs (hunk 4 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/AATFReturn/Mappings/ToViewModel/AddressUtilities.cs]: Tests created for this and tonnage utilities

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/AddAatfSiteRequestHandler.cs [hunk 7]
```diff
@@ -15,28 +17,33 @@ internal class AddAatfSiteRequestHandler : IRequestHandler<AddAatfSite, bool>
...
             this.offSiteDataAccess = offSiteDataAccess;
             this.genericDataAccess = genericDataAccess;
+            this.organisationDetailsDataAccess = organisationDetailsDataAccess;
         }
 
         public async Task<bool> HandleAsync(AddAatfSite message)
         {
             authorization.EnsureCanAccessExternalArea();
 
+            Country country = await organisationDetailsDataAccess.FetchCountryAsync(message.AddressData.CountryId);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/AddAatfSiteRequestHandler.cs]: Add an additional test to check that the country is retreived and the AatfAddress object country is property is set with the retrieved value

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/GetAatfSiteDataAccess.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+        public GetAatfSiteDataAccess(WeeeContext context, IGenericDataAccess genericDataAccess)
+        {
+            this.context = context;
+            this.genericDataAccess = genericDataAccess;
+        }
+
+        public async Task<List<AatfAddress>> GetAddresses(Guid aatfId, Guid returnId)
+        {
+            var weeeReusedId = (await genericDataAccess.GetManyByExpression<WeeeReused>(new WeeeReusedByAatfIdAndReturnIdSpecification(aatfId, returnId))).Last().Id;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/GetAatfSiteDataAccess.cs]: We can also write unit tests for this without checking the database... check ReturnSchemeDataAccessTests for examples

**Pull request merged**